simple/directed.go

```diff
@@ -120,7 +120,6 @@ func (g *DirectedGraph) RemoveNode(n graph.Node) {
 	}
 	delete(g.predecessors, n.ID())
 
-	g.maxID-- // Fun facts: even if this ID doesn't exist this still works!
```

Comment:
Yes, agreed.  Then it would also seem that we should move the addition to the freeMap to the else clause of the condition.  Can't do both (de-increment and add to freeMap).